### PR TITLE
Fix Optimistic Sends

### PR DESCRIPTION
**Pull request by AdityaSripal**
## Description

Companion PR to the following spec change: https://github.com/cosmos/ibc/pull/778

closes: #XXXX

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Comment by crodriguezvega** [comment]
Is this worth a simple manual test just as sanity check?

File: modules/apps/transfer/keeper/relay_test.go [hunk 1]
```diff
@@ -345,12 +345,14 @@ func (suite *KeeperTestSuite) TestOnTimeoutPacket() {
 
 	testCases := []struct {
 		msg      string
-		malleate func()
+		malleate func(suite *KeeperTestSuite)
```
Comment (damiannolan) [inline modules/apps/transfer/keeper/relay_test.go:348]: Does `suite` need to be passed here? I thought it was accessible regardless?

**Review by seantking — APPROVED** [review]
A manual test would be nice. Soon e2e tests :D

**Comment by AdityaSripal** [comment]
Need to determine viability of this feature before moving forward since we cannot do one-shot optimistic sends

**Comment by crodriguezvega** [comment]
Closing this since optimistic sends will need a more thorough fix overall.

**Pull request closed without merging**